resources/party.ts

```diff
@@ -155,4 +171,46 @@ export default class PartyTracker {
   nameFromId(id: string): string | undefined {
     return this.idToName_[id];
   }
+
+  paramObjectFromName(
+    name: string,
+    defaultValue: keyof PartyMemberParamObject,
+    playerNicks: { [name: string]: string },
+  ): PartyMemberParamObject | undefined {
+    const partyMember = this.details.find((member) => member.name === name);
+    let ret: PartyMemberParamObject;
+    if (!partyMember) {
+      // If we can't find this party member for some reason, use some sort of default
+      ret = {
+        id: '???',
+        job: 'NONE',
+        role: '???',
```

Comment:
Given the fallbacks, do you think it would make sense to have all of these be optional other than the name/shortName. That way if somebody has a trigger that says `${player1.role}` and something has gone awry, they'll get player1's name instead of `???`.

Reply:
I was saying something different; there's only two states, either we know everything or we only know name. I'm suggesting to make all the fields optional and possibly undefined so that in the case where we only know name, we only need to fill in name and shortName. That way if somebody specifies a missing field (that would normally be filled in but maybe it's the OP bug so it's not) then instead of getting '???' or 'NONE' when that's not useful, they would at least get the name. In other words, only fill in real values and never use dummy values or said differently never add something to this object that you wouldn't want displayed in a trigger.

Reply:
Addressed this in 5121b9926bc94eacbdca86b4616cdcc6645fb511